File: memento/memento.py
class Memento:
    def __init__(self, estado):
        self.estado = estado


class Editor:
    def __init__(self):
        self.contenido = ""

    def escribir(self, texto):
        self.contenido = texto

    def guardar(self):
        return Memento(self.contenido)

    def restaurar(self, memento):
        self.contenido = memento.estado

File: memento/memento.js
class Memento {
    constructor(estado) {
        this.estado = estado;
    }
}

class Editor {
    constructor() {
        this.contenido = "";
    }

    escribir(texto) {
        this.contenido = texto;
    }

    guardar() {
        return new Memento(this.contenido);
    }

    restaurar(memento) {
        this.contenido = memento.estado;
    }
}
